### PR TITLE
feat(style): adding theme key support for more style props

### PR DESCRIPTION
**Pull request by zchenwei**
#### Description of changes

This PR adds theme key for `space`, `radii`, `fontSizes` and `fontWeights`. 

#### Issue #, if available

An add-on to #2054 

#### Description of how you validated changes

Test it out locally

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/primitives/Link/__tests__/Link.test.tsx [hunk 1]
```diff
@@ -87,7 +87,7 @@ describe('Link: ', () => {
     expect(link.nodeName).toBe('P');
   });
 
-  it('can apply styling via props', async () => {
+  it.only('can apply styling via props', async () => {
```
Comment (zchenwei) [inline packages/react/src/primitives/Link/__tests__/Link.test.tsx:90]: Oh shoot, I should remove `only`. Will update later

Comment (wlee221) [reply packages/react/src/primitives/Link/__tests__/Link.test.tsx:90]: 😈

File: packages/react/src/primitives/types/style.ts [hunk 2]
```diff
@@ -47,52 +53,84 @@ export interface BaseStyleProps extends FlexItemStyleProps, GridItemStyleProps {
...
-  paddingBlockEnd?: ResponsiveStyle<StyleToken<Property.MarginBlockEnd>>;
-  paddingBlockStart?: ResponsiveStyle<StyleToken<Property.MarginBlockStart>>;
-  paddingBottom?: ResponsiveStyle<StyleToken<Property.PaddingBlockEnd>>;
-  paddingInlineEnd?: ResponsiveStyle<StyleToken<Property.MarginInlineEnd>>;
-  paddingInlineStart?: ResponsiveStyle<StyleToken<Property.MarginInlineStart>>;
-  paddingLeft?: ResponsiveStyle<StyleToken<Property.PaddingInlineStart>>;
-  paddingRight?: ResponsiveStyle<StyleToken<Property.PaddingInlineEnd>>;
-  paddingTop?: ResponsiveStyle<StyleToken<Property.PaddingBlockStart>>;
+  paddingBlockEnd?: ResponsiveStyle<
+    SpaceKeys<StyleToken<Property.MarginBlockEnd>>
```
Comment (joebuono) [inline packages/react/src/primitives/types/style.ts:102]: `PaddingBlockEnd`

File: packages/react/src/primitives/types/style.ts [hunk 3]
```diff
@@ -47,52 +53,84 @@ export interface BaseStyleProps extends FlexItemStyleProps, GridItemStyleProps {
...
+  >;
+  paddingBlockStart?: ResponsiveStyle<
+    SpaceKeys<StyleToken<Property.MarginBlockStart>>
```
Comment (joebuono) [inline packages/react/src/primitives/types/style.ts:105]: `PaddingBlockStart`

File: packages/react/src/primitives/types/style.ts [hunk 4]
```diff
@@ -47,52 +53,84 @@ export interface BaseStyleProps extends FlexItemStyleProps, GridItemStyleProps {
...
+  >;
+  paddingBottom?: ResponsiveStyle<
+    SpaceKeys<StyleToken<Property.PaddingBlockEnd>>
+  >;
+  paddingInlineEnd?: ResponsiveStyle<
+    SpaceKeys<StyleToken<Property.MarginInlineEnd>>
```
Comment (joebuono) [inline packages/react/src/primitives/types/style.ts:111]: `PaddingInlineEnd`

File: packages/react/src/primitives/types/style.ts [hunk 5]
```diff
@@ -47,52 +53,84 @@ export interface BaseStyleProps extends FlexItemStyleProps, GridItemStyleProps {
...
+  >;
+  paddingInlineStart?: ResponsiveStyle<
+    SpaceKeys<StyleToken<Property.MarginInlineStart>>
```
Comment (joebuono) [inline packages/react/src/primitives/types/style.ts:114]: `PaddingInlineStart`

**Review by joebuono — APPROVED** [review]
Just a few changes, otherwise looks good!

**Review by wlee221 — APPROVED** [review]
🧙

**Comment by zchenwei** [comment]
> Just a few changes, otherwise looks good!

Those are style prop names. I don't think we should capitalize them

**Pull request merged**